src/main/scala/fm/http/client/Response.scala

```diff
@@ -54,10 +54,14 @@ sealed abstract class Response(response: HttpResponse) extends Closeable {
     import Response.CharsetRegex
 
     response.headers().get("Content-Type").toBlankOption.flatMap { contentType: String =>
-      for {
+      val contentTypeCharset: Option[Charset] = for {
         CharsetRegex(charset) <- CharsetRegex.findFirstIn(contentType)
         if charsetIsSupported(charset)
       } yield Charset.forName(charset)
+
+      val mimeTypeCharset: Option[Charset] = MimeTypes.mimeTypeToCharset.get(contentType)
+
+      contentTypeCharset orElse mimeTypeCharset
```

Comment:
Shouldn't this just be:
```scala
contentTypeCharset orElse MimeTypes.mimeTypeToCharset.get(contentType)
```
There is no reason to do the `MimeTypes.mimeTypeToCharset` lookup if `contentTypeCharset` is defined.

Reply:
I originally had that way but thought it was more readable the other, I can swap it to be more performant